From 0418e593e26ee341e9732d6915d14d5385a65b8f Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Sun, 4 Jan 2015 17:47:29 +0800
Subject: [PATCH 201/210] ov2722:revert W/A:33062,and correct one register
 setting

original setting would work well on BYT, but hang
on CHV when 480p video case
and this also fixup regression issue: 14703

Change-Id: I9eaa0bb940904a9c3a57c47d96a0c4619e2ec613
Tracked-On: https://jira01.devtools.intel.com/i#browse/IMINAN-14703
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/34851
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Fang Wang <fang.c.wang@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/i2c/ov2722.h              | 26 +++++++++++-----------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h
index 5336dd1..283a037 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h
@@ -434,7 +434,7 @@ static struct ov2722_reg const ov2722_480P_30fps[] = {
 	{OV2722_8BIT, 0x3a0d, 0x00},
 	{OV2722_8BIT, 0x3a0e, 0x00},
 	{OV2722_8BIT, 0x4520, 0x0a},
-	{OV2722_8BIT, 0x4837, 0x29},
+	{OV2722_8BIT, 0x4837, 0x1b},
 	{OV2722_8BIT, 0x3000, 0xff},
 	{OV2722_8BIT, 0x3001, 0xff},
 	{OV2722_8BIT, 0x3002, 0xf0},
@@ -1121,18 +1121,18 @@ static struct ov2722_reg const ov2722_720p_30fps[] = {
 struct ov2722_resolution ov2722_res_preview[] = {
 	{
 		.desc = "ov2722_480P_30fps",
-		.width = 1632,
-		.height = 1092,
+		.width = 736,
+		.height = 496,
 		.fps = 30,
-		.pix_clk_freq = 85,
+		.pix_clk_freq = 73,
 		.used = 0,
-		.pixels_per_line = 2260,
-		.lines_per_frame = 1244,
+		.pixels_per_line = 2048,
+		.lines_per_frame = 1184,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
-		.regs = ov2722_1632_1092_30fps,
+		.regs = ov2722_480P_30fps,
 	},
 
 	{
@@ -1241,18 +1241,18 @@ struct ov2722_resolution ov2722_res_video[] = {
 	},
 	{
 		.desc = "ov2722_480P_30fps",
-		.width = 1632,
-		.height = 1092,
+		.width = 736,
+		.height = 496,
 		.fps = 30,
-		.pix_clk_freq = 85,
+		.pix_clk_freq = 73,
 		.used = 0,
-		.pixels_per_line = 2260,
-		.lines_per_frame = 1244,
+		.pixels_per_line = 2048,
+		.lines_per_frame = 1184,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
-		.regs = ov2722_1632_1092_30fps,
+		.regs = ov2722_480P_30fps,
 	},
 	{
 		.desc = "ov2722_720p_30fps",
-- 
1.9.1

